core-rust/state-manager/src/metrics.rs

```diff
@@ -330,6 +344,202 @@ impl CommittedTransactionsMetrics {
     }
 }
 
+impl ProtocolMetrics {
+    pub fn new(registry: &Registry, initial_protocol_state: &ProtocolState) -> Self {
+        let instance = Self {
+            protocol_update_readiness_ratio: GaugeVec::new(
+                opts(
+                    "protocol_update_readiness_ratio",
+                    "A ratio of supporting stake to total stake in the current validator set.",
+                ),
+                &["readiness_signal_name"],
+            )
+            .registered_at(registry),
+            pending_update_threshold_required_ratio: GaugeVec::new(
+                opts(
+                    "protocol_update_pending_threshold_required_ratio",
+                    "Required readiness ratio for the given protocol update threshold.",
+                ),
+                &["protocol_version_name", "readiness_signal_name", "threshold_index"],
+            )
+                .registered_at(registry),
+            pending_update_threshold_required_consecutive_epochs: IntGaugeVec::new(
+                opts(
+                    "protocol_update_pending_threshold_required_consecutive_epochs",
+                    "Required number of consecutive epochs of support for the given protocol update threshold.",
+                ),
+                &["protocol_version_name", "readiness_signal_name", "threshold_index"],
+            )
+                .registered_at(registry),
+            pending_update_threshold_matching_consecutive_epochs: IntGaugeVec::new(
```

Comment:
Ohh nice idea! :)

We should have a `pending_update_threshold_required_consecutive_epochs` as well as a `pending_update_threshold_current_consecutive_epochs` so we can draw a graph and check the current vs the required.

Reply:
yup, that's the "current" (I'll rename it)